ts/common.ts

```diff
@@ -0,0 +1,26 @@
+function copyCodePanel(id) {
+    let temp: any = document.getElementById(id);
```

Comment:
I don't like the use of `any` here; can we try checking for the existence of `textContent`, and if `textContent` doesn't exist we throw?